tests/langchain/test_langchain_tracer.py

```diff
@@ -508,6 +510,8 @@ def worker_function(worker_id):
         tracer.on_chain_start(
             {}, {"input": "test input"}, run_id=chain_run_id, name=f"chain_{worker_id}"
         )
+        # wait for a random time (0.5 ~ 1s) to simulate real-world scenario
+        time.sleep(random.random() / 2 + 0.5)
```

Comment:
This is why the issue was not captured before. Since `on_chain_start` and `on_chain_end` is executed in rapid succession, the `_root_active_id` is not polluted by other threads. However, the test fails after adding a little time gap between two.